### PR TITLE
Riot Boss

### PR DESCRIPTION
**Pull request by kblack89**
## Describe your changes

Adding a riot boss that will use a shield and napalm attacks so the player cant always damage them and will force the player to avoid areas hit by napalm.

## Issue number and link

#107 

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

**Comment by kblack89** [comment]
This is ready for review, please give feedback on pacing of combat for boss fight. The RiotBoss scene is the one to test out on, adjust wave number and time as you wish.

File: Assets/Scripts/EnemyScripts/Bosses/RiotBossBehaviour.cs [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    {
+        GameObject newBullet = Instantiate(bullet, bulletPosition.position, Quaternion.identity);
+        newBullet.GetComponent<BossBullet>().InitializeBullet(player, Damage, false);
+        SFXManager.Instance.PlaySFX("EnemyShoot");
+        attackCooldown = attackInterval; 
+    }
+
+    private IEnumerator SpawnNapalmBomb()
+    {
+        while (true)
```
Comment (AlexReidNZ) [inline Assets/Scripts/EnemyScripts/Bosses/RiotBossBehaviour.cs:107]: This coroutine, as well as the spawnShield one beneath need to only happen when the game is playing.
you can just add 
``if (GameSettings.gameState != GameState.InGame) { return; }``
At the start of both of them.

**Review by AlexReidNZ — CHANGES_REQUESTED** [review]
This boss works well. The napalm attack and the shield don't stop when the game is paused. Fix is in a seperate comment. Bleed bug is fixed.

**Review by AlexReidNZ — APPROVED** [review]
Looks good now, all found bugs appear to be fixed. Ready for merging.

**Review by Loerad — APPROVED** [review]
Looks fine. Maybe something with piercing bullets going through the shield? Not mandatory if you think its better without.

**Pull request merged**